connectors/memory/memory.go

```diff
@@ -486,8 +435,15 @@ func (c *Connector) Shutdown() error {
 	return nil
 }
 
+// NewConnector creates a new in-memory connector
+func NewConnector() *Connector {
```

Comment:
One interesting problem here is that in the future I'd like to allow someone to call a method in the memory connector that isn't part of the normal Connector interface -- for example, to clear or allocate a request level cache, LRU expire some items, etc. In that case, if you need the concrete implementation, you would want this method.

Arguably, as of right now, nobody should be calling this except perhaps tests, so it could be unexported. The correct way to get one of these is to call dosa.GetConnector.